cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -0,0 +1,394 @@
+
+Here is why filtering URLs is hard at the Application layer:
+
+* It implies that the application must be able to detect, at the code level, that the provided IP (V4 + V6) is not part of the official [private networks ranges](https://en.wikipedia.org/wiki/Private_network) including also *localhost* and *IPv4/v6 Link-Local* addresses. Not every SDK provides a built-in feature for this kind of verification, and leaves the handling up to the developer understand all of its pitfalls and possible values, which makes it a demanding task.
+* Same remark for domain name: The company must maintains a list of all internal domain names and provide a centralized service to allow an application to verify if a provided domain name is an internal one. For this verification, an internal DNS resolver can be queried by the application but this internal DNS resolver must not resolve external domain names.
+
+### Available protections
+
+We take the same assumption than for the case n°1 regarding the application user input data needed and the *defense in deph* principle.
+
+#### Application layer
+
+Like for the case [n°1](Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md#case-1---application-can-send-request-only-to-identified-and-trusted-applications), it is assumed that the `IP Address` or `domain name` is required to create the request that will be sent to the *TargetApplication*.
+
+The first validation on the input data presented in the case [n°1](Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md#application-layer) on the 3 types of data will be the same for this case **BUT the second validation will differ**. Indeed, here we must use the blacklist approach.
+
+> **Regarding the proof of legitimacy of the request**: The *TargetedApplication* that will receive the request must generate a random token (ex: alphanumeric of 20 characters) that is expected to be passed by the caller (in body via a parameter for which the name is also defined by the application itself and only allow characters set `[a-z]{1,10}`) to perform a valid request. The receiving endpoint must only accept HTTP POST requests.
+
+**Validation flow (if one the validation steps fail then the request is rejected):**
+
+1. The application receive the IP address or domain name of the *TargetedApplication* and apply the first validation on the input data using the libraries/regex mentioned in this [section](Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md#application-layer).
+2. The second validation is applied against the IP address or domain name of the *TargetedApplication* using the following blacklist approach:
+    * For IP address: 
+        * The application verify that it is a public one (see the hint provided in the next paragraph with the python code sample).
```

Comment:
verifies instead of verify

Reply:
Taken in account in the coming commit.
To re-check because I have reworded all this paragraph in order to remove all the `we`.

Reply:
Done